src/gui.rs

```diff
@@ -394,195 +375,26 @@ impl Sam {
-                ui.horizontal_centered(|ui| {
-                    self.top_bar(ui, false);
-                });
-            });
-        } else {
-            egui::TopBottomPanel::top("topbar").min_height(20.0).max_height(300.0).show(ctx, |ui| {
-                ui.set_enabled(!self.archive_window_open);
-                CollapsingHeader::new("Status & Controls").default_open(false).show(ui, |ui| {
-                    self.top_bar(ui, true);
-                    ui.add_space(10.0);
-                });
-            });
-        }
+        create_header(self, ctx);
 
         // Bottom status bar
-        egui::TopBottomPanel::bottom("bottombar").min_height(30.0).show(ctx, |ui| {
-            ui.set_enabled(!self.archive_window_open);
-            ui.horizontal_centered(|ui| {
-                // Give the data source some space on the left ...
-                ui.horizontal_centered(|ui| {
-                    ui.set_width(ui.available_width() / 2.0);
-                    self.data_source.status_bar_ui(ui);
-                });
-
-                // ... and the current tab some space on the right.
-                ui.allocate_ui_with_layout(ui.available_size(), Layout::right_to_left(Align::Center), |ui| {
-                    #[cfg(not(target_arch = "wasm32"))]
-                    ui.add_enabled_ui(!self.data_source.is_log_file(), |ui| {
-                        if ui.button("⏮  Reset").clicked() {
-                            self.data_source.reset();
-                        }
-                    });
-
-                    #[cfg(not(target_arch = "wasm32"))]
-                    ui.separator();
-
-                    match self.tab {
-                        GuiTab::Launch => {}
-                        GuiTab::Plot => self.plot_tab.bottom_bar_ui(ui, self.data_source.as_mut()),
-                        GuiTab::Configure => {}
-                    }
-                });
-            });
-        });
+        create_bottom_status_bar(self, ctx);
 
-        // Everything else. This has to be called after all the other panels
-        // are created.
-        egui::CentralPanel::default().show(ctx, |ui| {
-            ui.set_width(ui.available_width());
-            ui.set_height(ui.available_height());
-            ui.set_enabled(!self.archive_window_open);
-
-            match self.tab {
-                GuiTab::Launch => {}
-                GuiTab::Plot => self.plot_tab.main_ui(ui, self.data_source.as_mut()),
-                GuiTab::Configure => {
-                    let changed = self.configure_tab.main_ui(ui, self.data_source.as_mut(), &mut self.settings);
-                    if changed {
-                        self.data_source.apply_settings(&self.settings);
-                        self.plot_tab.apply_settings(&self.settings);
-                    }
-                }
-            }
-        });
+        // Everything else. This has to be called after all the other panels are created.
+        create_body(self, ctx);
```

Comment:
I think this might be general enough to stay in here. See reasoning above regarding keeping all the `match self.tab { ... }` stuff in one place.